nrf52-hal-common/src/temp.rs

```diff
@@ -0,0 +1,61 @@
+//! Temperature sensor interface.
+
+use fpa::I30F2;
+use nb;
+use crate::target::TEMP;
+
+/// Integrated temperature sensor.
+pub struct Temp(TEMP);
+
+impl Temp {
+    /// Creates a new `Temp`, taking ownership of the temperature sensor's register block.
+    pub fn new(raw: TEMP) -> Self {
+        Temp(raw)
+    }
+
+    /// Starts a new measurement and blocks until completion.
+    ///
+    /// If a measurement was already started, it will be canceled.
+    pub fn measure(&mut self) -> I30F2 {
+        self.stop_measurement();
+        self.start_measurement();
+
+        while self.0.events_datardy.read().bits() == 0 {}
+
+        self.0.events_datardy.reset(); // clear event
+        let raw = self.0.temp.read().bits();
+        I30F2::from_bits(raw as i32)
```

Comment:
Why not call `read` in a loop instead? This seems to be the same code, basically.

Reply:
Yes, this can just use `nb::block!`

Reply:
Done :)